docs/modules/ROOT/pages/how-tos/testing/integration/accessing-private-repositories.adoc

```diff
@@ -0,0 +1,71 @@
+= Access Pipelines and Tasks in Private Repositories
+
+Integration Test Scenarios point to their corresponding pipeline with a link:https://tekton.dev/docs/pipelines/git-resolver/[Git Resolver]. The Git Resolver schema is inherited from Tekton and functions in the same way. Users should create an access token for their repository in link:https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens[Github] or link:https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html[Gitlab] then link:../configuring/creating-secrets.adoc#creating-source-control-secrets[store that token in a secret] in their namespace.
+
+Once this is done, the Resolver should be updated with the fields `token` and `tokenKey`, which provide the name of the secret and the key within the `data` section of that secret in which the access token is stored.
+
+== Example of Resolver in IntegrationTestScenario
+
+[source,yaml]
+.its.yaml
+
+----
+apiVersion: appstudio.redhat.com/v1beta2
+kind: IntegrationTestScenario
+metadata:
+  name: example-pass
+  namespace: default
+spec:
+  application: application-sample
+  contexts:
+    - description: Application testing
+      name: application
+  resolverRef:
+    resolver: git
+    params:
+      - name: url
+        value: https://github.com/konflux-ci/sample-private-repo
+      - name: revision
+        value: main
+      - name: pathInRepo
+        value: pipelines/integration_pipeline_pass.yaml
+      - name: token
+        value: ${SECRET_NAME}
+      - name: tokenKey
+        value: ${SECRET_KEY}
+----
+
+== Example of Resolver in Pipeline
+
+This is an example of how tasks in private repos can be accessed with the Git Resolver using the same method as above.
+
+[source,yaml]
+.pipeline.yaml
+
+----
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: deploy-and-test
+spec:
+  description: |
+    Demonstrates the use of Git Resolvers to access tasks in private repos
+  tasks:
+    - name: parse-metadata
+        taskRef:
+          resolver: git
+          params:
+            - name: url
+              value: https://github.com/konflux-ci/sample-private-repo
```

Comment:
I don't think that this will work as intended. When URL is defined tekton uses  Anonymous Git and tokens are unused. Git API method must be used instead.

https://github.com/tektoncd/pipeline/blob/084423328a632ecfb0e339d761f02c9dbefe932e/pkg/resolution/resolver/git/resolver.go#L132

https://tekton.dev/docs/pipelines/git-resolver/#task-resolution-with-a-custom-token-to-a-custom-scm-provider

Reply:
Good catch, fixed